client/src/redux/sagas/users.sagas.js

```diff
@@ -0,0 +1,46 @@
+import { put, call, debounce } from 'redux-saga/effects';
+import fetch from 'cross-fetch';
+import Cookies from 'js-cookie';
+
+export async function fetchUserSearch(q) {
+  const url = `/api/users/?q=${q}`;
+  const response = await fetch(url, {
+    method: 'GET',
+    headers: {
+      'X-CSRFToken': Cookies.get('csrftoken'),
+      'Content-Type': 'application/json'
+    },
+    credentials: 'same-origin'
+  });
+  if (response.status === 404) {
+    return [];
+  }
+  if (response.status === 200) {
+    const result = await response.json();
+    return result;
+  }
+  throw Error('Users search error');
+}
+
+export function* userSearch(action) {
```

Comment:
@owaisj Any overlap here with what you're doing?

Reply:
We had a conversation and chose not to reconcile/refactor the two versions into one component at this time due to his search going to TAS and my search going to the local Django DB. I think there's a way we could make a view-only component and feed in the search results and pass back the onchange events when a user types stuff, but it may be an extremely thin component. @owaisj  your thoughts?

Reply:
Would this component just use different callbacks? 

I agree that it would be thin

Reply:
Something like that. It's going to be tricky, especially if the user objects in TAS don't match the user objects in Django because then you'd have to also pass down view components, which is gross. I think maybe for now if you want to steal this code, re-engineer it for Allocations and then we sit down and reconcile the two of them later that may be best.